src/corecel/math/Algorithms.hh

```diff
@@ -123,22 +123,22 @@ struct Less<void>
  * This is useful for calls to \c std::all_of .
  */
 template<class T = void>
-struct LogicalTrue
+struct Identity
 {
-    CELER_CONSTEXPR_FUNCTION bool operator()(T const& value) const noexcept
+    CELER_CONSTEXPR_FUNCTION const T&& operator()(T const& value) const noexcept
     {
-        return static_cast<bool>(value);
+        return std::forward<T const>(value);
     }
 };
 
-//! Specialization of LogicalTrue with template deduction
+//! Specialization of Identity with template deduction
 template<>
-struct LogicalTrue<void>
+struct Identity<void>
 {
     template<class T>
-    CELER_CONSTEXPR_FUNCTION bool operator()(T const& value) const noexcept
+    CELER_CONSTEXPR_FUNCTION const T&& operator()(T const& value) const noexcept
     {
-        return static_cast<bool>(value);
+        return std::forward<T const>(value);
     }
```

Comment:
In C++20 `identity` isn't templated on anything; only its `operator()` is. Should we do the same (mirroring the inconsistency between the identity and other functional unary operators?)